x-pack/plugins/fleet/common/services/full_agent_policy_to_yaml.ts

```diff
@@ -25,23 +25,45 @@ const POLICY_KEYS_ORDER = [
   'meta',
   'input',
   'download',
+  'signed',
 ];
 
 export const fullAgentPolicyToYaml = (policy: FullAgentPolicy, toYaml: typeof safeDump): string => {
-  return toYaml(policy, {
+  const yaml = toYaml(policy, {
     skipInvalid: true,
-    sortKeys: (keyA: string, keyB: string) => {
-      const indexA = POLICY_KEYS_ORDER.indexOf(keyA);
-      const indexB = POLICY_KEYS_ORDER.indexOf(keyB);
-      if (indexA >= 0 && indexB < 0) {
-        return -1;
-      }
-
-      if (indexA < 0 && indexB >= 0) {
-        return 1;
-      }
-
-      return indexA - indexB;
-    },
+    sortKeys: _sortYamlKeys,
   });
+
+  if (!policy?.secret_references?.length) return yaml;
+
+  return _formatSecrets(policy.secret_references, yaml);
 };
+
+function _sortYamlKeys(keyA: string, keyB: string) {
+  const indexA = POLICY_KEYS_ORDER.indexOf(keyA);
+  const indexB = POLICY_KEYS_ORDER.indexOf(keyB);
+  if (indexA >= 0 && indexB < 0) {
+    return -1;
+  }
+
+  if (indexA < 0 && indexB >= 0) {
+    return 1;
+  }
+
+  return indexA - indexB;
+}
+
+function _formatSecrets(
+  secretRefs: NonNullable<FullAgentPolicy['secret_references']>,
+  ymlText: string
+) {
+  let formattedText = ymlText;
+  const secretIds = secretRefs.map((ref) => ref.id);
+
+  secretIds.forEach((secretId, idx) => {
+    const regex = new RegExp(`\\$co\\.elastic\\.secret\\{${secretId}\\}`, 'g');
+    formattedText = formattedText.replace(regex, `\${SECRET_${idx}}`);
```

Comment:
It is not possible to use the same `secretId` multiple times in one policy, right? Asking because replace only replaces the first match.